### PR TITLE
Improve tests for history visibility

### PR DESCRIPTION
**Pull request by richvdh**
This tests the functionality modifications in https://github.com/matrix-org/synapse/pull/587.

De-conflate the ~~anonymous~~guest access and world_readable tests; move the
history visibility tests together into 30history-visibility.pl and extend them
to cover world_readable. Put the tests for guest_access in 13guestaccess.pl.

### CODE REVIEW

File: tests/30rooms/13guestaccess.pl [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+
+   do => sub {
+      my ( $user, $room_id, $guest_user ) = @_;
+
+      matrix_set_room_guest_access( $user, $room_id, "can_join" )
+      ->then( sub {
+         matrix_join_room( $guest_user, $room_id )
+      })->then( sub {
+         matrix_send_room_text_message( $guest_user, $room_id, body => "sup" );
+      })->then(sub {
```
Comment (leonerd) [inline tests/30rooms/13guestaccess.pl]: Whitespace `( sub {`

File: tests/30rooms/13guestaccess.pl [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+         matrix_get_room_messages( $user, $room_id, limit => 1 )->then( sub {
```
Comment (leonerd) [inline tests/30rooms/13guestaccess.pl]: This `->then` can be outdented a level

**Comment by richvdh** [comment]
@leonerd: PTAL

**Comment by leonerd** [comment]
Now with a few other minor bits of syntax/whitespace neatening.

LGTM

**Comment by richvdh** [comment]
thanks for the cleanups.

**Pull request merged**